tests/macros/apiref.test.js

```diff
@@ -0,0 +1,262 @@
+        id: 87763,
+        review_tags: [],
+        slug: 'Web/API/Gamepad/buttons',
+        uuid: '6b6f5f4d-81cb-4b65-81dc-88dc94bbd494',
+        title: 'Gamepad.buttons',
+        url: '/en-US/docs/Web/API/Gamepad/buttons',
+        modified: '2019-01-16T13:03:39.199620',
+        label: 'Gamepad.buttons',
+        localization_tags: [],
+        last_edit: '2018-07-20T12:21:09.854338',
+        sections: [
+            {
+                id: 'Quick_Links',
+                title: null
+            },
+            {
+                id: 'Syntax',
+                title: 'Syntax'
+            },
+            {
+                id: 'Example',
+                title: 'Example'
+            },
+            {
+                id: 'Value',
+                title: 'Value'
+            },
+            {
+                id: 'Specifications',
+                title: 'Specifications'
+            },
+            {
+                id: 'Browser_compatibility',
+                title: 'Browser compatibility'
+            },
+            {
+                id: 'sect1',
+                title: null
+            },
+            {
+                id: 'sect2',
+                title: null
+            },
+            {
+                id: 'sect3',
+                title: null
+            },
+            {
+                id: 'sect4',
+                title: null
+            },
+            {
+                id: 'sect5',
+                title: null
+            },
+            {
+                id: 'sect6',
+                title: null
+            },
+            {
+                id: 'Legend',
+                title: 'Legend'
+            },
+            {
+                id: 'See_also',
+                title: 'See also'
+            }
+        ]
+    }
+];
+
+function checkResult(html, locale) {
+    // Lint the HTML
+    expect(lintHTML(html)).toBeFalsy();
+    const dom = JSDOM.fragment(html);
+    // Check that all links reference the proper locale or use https
+    const num_total_links = dom.querySelectorAll('a[href]').length;
+    const num_valid_links = dom.querySelectorAll(
+        `a[href^="/${locale}/"], a[href^="https://"]`
+    ).length;
+    expect(num_valid_links).toEqual(num_total_links);
+}
+
+describeMacro('APIRef', function() {
+    beforeEachMacro(function(macro) {
+        // Mock calls to MDN.subpagesExpand
+        macro.ctx.page.subpagesExpand = jest.fn(() => {
+            return GAMEPAD_SUBPAGES_EXPAND;
+        });
+    });
+
+    for (const locale of ['en-US', 'fr', 'ja']) {
+        itMacro(`with locale ${locale}`, function(macro) {
+            macro.ctx.env.locale = locale;
+            macro.ctx.env.slug = 'Web/API/Gamepad';
+            return macro.call('Gamepad API').then(function(result) {
+                checkResult(result, locale);
```

Comment:
You've got two different tests combined into a single checkResult() here.

I think it would be clearer (though slower) to have two separate tests named tests:

```
describeMacro(..., ()=>{
   beforeEachMacro(macro => {
   });

    it('produces valid html', () => {
    });

    it('produces locale-specific links', () => {
    });
});
```

This would mean duplicating the `for locale of ` loop inside each it() function, and doubling the number of macro calls, but it would give you better reporting when failures occur.